File: python-solution/src/search_sub_domain.py
import re

def search_sub_domain(sub_domain, data_array):
  filtered_array = []

  for array in data_array:
    [_, domain] = array

    if len(domain) < len(sub_domain):
      continue

    if domain == sub_domain:
      filtered_array.append(array)

    pattern = rf"\.{sub_domain}$"

    if re.search(pattern, domain):
      filtered_array.append(array)

  counter = 0
  for array in filtered_array:
    [visitors, _] = array
    counter += int(visitors, 10)

  return [counter, sub_domain]

File: golang-solution/src/searchSubDomain.go
package src

import (
    "regexp"
    "strings"
    "strconv"
)

func searchSubDomain(subDomain string, data [][]string) []string {
    filteredSlice := [][]string{}

    for _, element := range data {
        var domain string = element[1]

        if len(domain) < len(subDomain) {
            continue
        }

        if domain == subDomain {
            filteredSlice = append(filteredSlice, element)
        }

        s := []string{"\\.", subDomain, "$"}
        pattern, _ := regexp.Compile(strings.Join(s, ""))

        if pattern.MatchString(domain) {
            filteredSlice = append(filteredSlice, element)
        }
    }

    counter := 0
    for _, element := range filteredSlice {
        visitors, _ := strconv.Atoi(element[0])
        counter += visitors
    }
    result := []string{strconv.Itoa(counter), subDomain}

    return result
}
